### PR TITLE
389 post timeout

### LINKED ISSUE #389: [389] Post timeout
**Current state**
Users are able to spam posts, this should be prohibited

**Expected state**
A timeout should be added to prevent users from spamming posts

**TO DO**
- [x] Add attribute to backend for timeout length
- [x] Add check to backend for spam posts
- [x] Add check and warning to frontend when adding posts before timeout completes
- [ ] Add message when user is not allowed to post yet

**Branch**
[389-post-timeout](https://github.com/AjobK/Seaqull/tree/389-post-timeout)

### PR DESCRIPTION
**Pull request by MrGooss**
closes #389 

<br />

 **What is it supposed to do**  
Adds a timeout for users trying to spam posts.

Used the branch from issue #386 to get a toast when someone tries to spam posts, duration of the timeout can be updated in the .env file
image

<br />

 **How can we test your branch**  
Create a post and try to create another post directly after. This should display an error and not create the post.

<br />

 **Additional notes**  
UPDATE THE ENV FILE

<br />

**1. General checklist**
- [x] a. Tested happy flow.
- [x] b. Tested unhappy flow.
- [x] c. No unexpected exceptions.
- [x] d. No code with high complexity. (Big O)
- [x] e. EOF newline.
- [x] f. No linter warnings or errors.

<br />

**2. If relevant, front-end checklist**
- [x] a. Responsive. (Desktop all the way to mobile 320px)
- [x] b. Using SCSS variables where relevant.
- [x] c. Relevant class names
- [x] d. No modified global SCSS properties, except if necessary and fully tested the impact.
- [x] e. Checked [caniuse](https://caniuse.com) for more modern CSS properties.

<br />

**3. If relevant, back-end checklist**
- [x] a. No unhandled promise rejections.
- [x] b. Seeders still functional.
- [x] c. No primary-key violations.
- [x] d. Normalization, at least [3NF](https://en.wikipedia.org/wiki/Database_normalization).
- [x] e. Used guards (middleware) where necessary and tested if route is guarded.

<br />

**4. If relevant, test these browsers**
- [x] a. Chrome
- [ ] b. Firefox
- [ ] c. Opera
- [ ] d. Edge
- [ ] e. Safari

---

**PR Rules**  
*1. Add relevant labels.*  
*2. Select at least two reviewers.*  
*3. Assign all reviewers allowed to merge the branch, including yourself.*  
*4. If relevant to the sprint, put related issue in 'Awaiting Approval' column on project board.*  
*5. Add to the corresponding [milestone](https://github.com/AjobK/Seaqull/milestones).*

<br />

**Reviewer info**  
*1. You can use aliases to refer to specific PR checklist items. For example 'Hey @JohnDoe, for this line please check 3.a and 1.d.'*  
*2. Where possible, please use the suggestions feature in github so the author has a better idea of the solution you had in mind.*   

<br />

> This form should be filled in by the author of this PR.

### CODE REVIEW

**Comment by AjobK** [comment]
It doesn't cause an error..?

image


I created like 7 posts in a row

File: backend/src/services/post.service.ts [hunk 1]
```diff
@@ -141,6 +141,11 @@ export class PostService {
   }
 
   public async createPost(createPostDTO: CreatePostDTO, file: Express.Multer.File, user: Account): Promise<string> {
+    if (!this.isAllowedToPost(user)) {
+      //TODO: add messaging system if not there yet?
```
Comment (AjobK) [inline backend/src/services/post.service.ts]: What does this mean?

**Review by AjobK — CHANGES_REQUESTED** [review]

File: backend/src/services/post.service.ts (hunk 1 above)
Comment (AjobK) [reply backend/src/services/post.service.ts]: Also, check my comment above. It doesn't seem to work

**Comment by MrGooss** [comment]
Make global config class with timeout setting

**Comment by AjobK** [comment]
@S-Goossens Try to make the TODO a little clearer next time. 'UPDATE ENV' lacks description. Anyway, I'm probably gonna remove the timeout from the env file as it's only used in one place right now. It also requires some configuration which we don't really need.

**Comment by AjobK** [comment]
The post timeout didn't work at first. You were comparing milliseconds, but this is not the total milliseconds of all time, only in the current second. So you would get 'random' msDiffs. Code now works.

**Review by AjobK — APPROVED** [review]

**Pull request merged**